src/lib/core/style/themeable.ts

```diff
@@ -0,0 +1,30 @@
+import {ElementRef, Input, Renderer} from '@angular/core';
+
+/** Material components can extend the MdThemeable class to add an Input that can
+ * developers use to switch palettes on the components. */
+export class MdThemeable {
+
+  /** Stored color for the themeable component. */
+  private _color: string;
+
+  constructor(private renderer: Renderer, private elementRef: ElementRef) {}
```

Comment:
Yeah makes sense. While doing that I ran into an issue with the child classes.

```
Types have separate declarations of a private property '_elementRef'.
```

Because of that the best solution is to just use the `protected` modifier here, and the child classes can just use the variables similar as before. (Just remove the modifiers in the constructor)